### ISSUE TITLE
Alerts & reports : Parser must be a string or character stream, not datetime

### ISSUE THREAD

**Issue by AbhishekOnedigital** [apache/superset#24691]
I created a report in a superset .my celery worker giving the parser error 

Report setting in screenshot 

![image](https://github.com/apache/superset/assets/17587331/f87c21b4-9be5-4d81-b848-4b4add5ed9aa)

Trace of error:
```python
Scheduling alert test eta: 2023-07-13 19:56:00
[2023-07-13 19:56:00,089: INFO/ForkPoolWorker-1] Scheduling alert test eta: 2023-07-13 19:56:00
[2023-07-13 19:56:00,094: ERROR/ForkPoolWorker-3] Task reports.execute[de8817cb-4be2-4eac-9622-4bc36af95594] raised unexpected: TypeError('Parser must be a string or character stream, not datetime')
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", line 477, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/superset/initialization/__init__.py", line 108, in __call__
    return task_base.__call__(self, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", line 760, in __protected_call__
    return self.run(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/superset/tasks/scheduler.py", line 81, in execute
    scheduled_dttm_ = parser.parse(scheduled_dttm)
  File "/usr/local/lib/python3.9/site-packages/dateutil/parser/_parser.py", line 1368, in parse
    return DEFAULTPARSER.parse(timestr, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/dateutil/parser/_parser.py", line 640, in parse
    res, skipped_tokens = self._parse(timestr, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/dateutil/parser/_parser.py", line 719, in _parse
    l = _timelex.split(timestr)         # Splits the timestr into tokens
  File "/usr/local/lib/python3.9/site-packages/dateutil/parser/_parser.py", line 201, in split
    return list(cls(s))
  File "/usr/local/lib/python3.9/site-packages/dateutil/parser/_parser.py", line 69, in __init__
    raise TypeError('Parser must be a string or character stream, not '
TypeError: Parser must be a string or character stream, not datetime
```
My superset config 

```python
FEATURE_FLAGS = {
    "ALERT_REPORTS": True,
    "THUMBNAILS": True,
    "THUMBNAILS_SQLA_LISTENERS": True,
}
REDIS_HOST = "localhost"
REDIS_PORT = "6379"

class CeleryConfig:
    broker_url = 'redis://%s:%s/0' % (REDIS_HOST, REDIS_PORT)
    imports = ('superset.sql_lab', "superset.tasks", "superset.tasks.thumbnails", )
    result_backend = 'redis://%s:%s/0' % (REDIS_HOST, REDIS_PORT)
    worker_prefetch_multiplier = 10
    task_acks_late = True
    task_annotations = {
        'sql_lab.get_sql_results': {
            'rate_limit': '100/s',
        },
        'email_reports.send': {
            'rate_limit': '1/s',
            'time_limit': 600,
            'soft_time_limit': 600,
            'ignore_result': True,
        },
    }
    beat_schedule = {
        'reports.scheduler': {
            'task': 'reports.scheduler',
            'schedule': crontab(minute='*', hour='*'),
        },
        'reports.prune_log': {
            'task': 'reports.prune_log',
            'schedule': crontab(minute=0, hour=0),
        },
    }
CELERY_CONFIG = CeleryConfig
```

Activating my celery command

```shell
nohup celery --app=superset.tasks.celery_app:app worker --pool=prefork -O fair -c 4  > /tmp/celery_first.out &
nohup celery --app=superset.tasks.celery_app:app beat   > /tmp/celery_second.out &
``` 

### Environment

- browser type and version:  Chrome 114
- superset version: `2.1.0`
- python version: `3.9` 
- celary version  : `5.3.1 (emerald-rush)`
- database for  superset - postgres 14

**Comment by soidamientrung** [comment]
I have the same error "TypeError: Parser must be a string or character stream, not datetime".

**Comment by Pooort** [comment]
The same one.

**Comment by Pooort** [comment]
Any ideas?

**Comment by Badgerr** [comment]
Getting the same error.

**Comment by sfirke** [comment]
Are you all using your own Celery instances, or the Celery containers that spin up with `docker-compose` ?

**Comment by mlkkkfriend** [comment]
I have the same problem

**Comment by mlkkkfriend** [comment]
python3 3.9.9

**Comment by cw1427** [comment]
The same issue

**Comment by ngurmen** [comment]
Hi @sfirke, I am having the same issue on a dev install from scratch using venv.

ubuntu 22.04
superset 2.1.1
python3.10.12
pip 23.2.1
celery 5.3.4 [ I downgraded 5.2.7 as it is working fine on 2.0.1 but I saw the same issue]

I am using the -B switch on celery running worker and beat in one go as follows:
```
celery --app=superset.tasks.celery_app:app worker \
       --pool=prefork --max-tasks-per-child=128 \
       -O fair -c 4 -E -B
```
Hope this helps.  Superset was installed from scratch but the metadata database was upgraded from 2.0.1.  Everything else seems to be working fine but it is not possible for us to upgrade without the alerts and reports working.

**Comment by ngurmen** [comment]
Quick update regarding the alerts and reports issue above.  I upgraded my dev install to superset 3.0.0.  Everything seems to be working as expected.  I tested the feature for both email and Slack.

ubuntu 22.04
--
superset 3.0.0
python 3.10.12
pip 23.2.1
celery 5.2.7
all added in venv

Additionally, I just had to install libecpg-dev package on ubuntu 22.04 for SQL Lab to properly give syntax alerts on queries.
